pkg/build/generator/generator.go

```diff
@@ -513,6 +513,11 @@ func (g *BuildGenerator) generateBuildFromConfig(ctx kapi.Context, bc *buildapi.
 				Kind: "DockerImage",
 				Name: runtimeImageName,
 			}
+			// We're using single secret for pulling builder and runtime images.
+			// If builder image doesn't need a secret, let's check whether runtime image needs it or not.
+			if build.Spec.Strategy.SourceStrategy.PullSecret == nil {
+				build.Spec.Strategy.SourceStrategy.PullSecret = g.resolveImageSecret(ctx, builderSecrets, build.Spec.Strategy.SourceStrategy.RuntimeImage, bc.Namespace)
+			}
```

Comment:
@bparees i remember a discussion on scrum from which i decided to leave it like this, i'm sorry - but i might have misunderstood back then how this works properly.